README.md

```diff
@@ -81,4 +112,148 @@ opa:
+|------------------------------------------------|---------|------------------------------------------------------------------------------------------|
+| `contentgrid.gateway.runtime-platform.enabled` | boolean | Enables runtime platform configuration                                                   |
+| `servicediscovery.enabled`                     | boolean | Enables service discovery in kubernetes                                                  |
+| `servicediscovery.namespace`                   | string  | Kubernetes namespace to perform service discovery in (default: `default`)                |
+| `servicediscovery.resync`                      | int     | Re-synchronisation interval of the service discovery informer in seconds (default: `60`) |
+
+Additionally, following properties are required to be configured:
+
+| Property                                           | Description                                                                                          |
+|----------------------------------------------------|------------------------------------------------------------------------------------------------------|
+| `opa.service.url`                                  | Must be configured to the shared OPA service that contains policies for all ContentGrid applications |
+| `contentgrid.gateway.jwt.signers.apps.active-keys` | The `apps` JWT Signer will be used for signing all requests to ContentGrid applications              |
+
+To discover ContentGrid applications, following kubernetes objects are required:
+
+- One `Service` object with labels:
+    - `app.kubernetes.io/managed-by=contentgrid`
+    - `app.contentgrid.com/service-type=api`
+    - `app.contentgrid.com/application-id`: Set to the application ID
+    - `app.contentgrid.com/deployment-id`: Set to the deployment ID
+
+- One or more `ConfigMap` or `Secret` objects with labels:
+    - `app.kubernetes.io/managed-by=contentgrid`
+    - `app.contentgrid.com/service-type=gateway`
+    - `app.contentgrid.com/application-id`: Set to the application ID
+
+The ContentGrid gateway will discover running ContentGrid applications using the `Service` object.
+There can be multiple deployments of the same Application ID running at the same time, these are disambiguated
+by the Deployment ID.
+
+Routing requests to a specific Deployment ID is not implemented yet.
+All requests will be routed to the same, arbitrarily selected Deployment ID.
+
+`Service`s can have annotations for configuration:
+
+| Annotation                             | Type   | Description                                                                  |
+|----------------------------------------|--------|------------------------------------------------------------------------------|
+| `authz.contentgrid.com/policy-package` | string | The OPA package that contains policies for evaluating access to this service |
```

Comment:
In the implementation this is a constant called: LABEL_POLICY_PACKAGE
I guess it is still an annotation, and not a label?

Reply:
It is an annotation, because it is not part of the _identity_ of the service.
Spring service discovery smashes labels and annotations together into one metadata map, so technically it wouldn't matter.

I will rename the constant to be `ANNOTATION_POLICY_PACKAGE`